tesseract_core/sdk/docker_client.py

```diff
@@ -110,26 +87,26 @@ def get(image_id_or_name: str | bytes, tesseract_only: bool = True) -> Image:
         if not image_id_or_name:
             raise ValueError("Image name cannot be empty.")
 
-        def _sanitize_image_id(image_id: str) -> str:
-            """Sanitize image id by removing sha256 prefix."""
-            if image_id.startswith("sha256:"):
-                return image_id[len("sha256:") :]
-            return image_id
-
-        image_id_or_name = _sanitize_image_id(image_id_or_name)
-        images = Images.list(tesseract_only=tesseract_only)
-
-        # Check for both name and id to find the image
-        # Tags may be prefixed by repository url
-        for image_obj in images:
-            if (
-                _sanitize_image_id(image_obj.id) == image_id_or_name
-                or _sanitize_image_id(image_obj.short_id) == image_id_or_name
-                or Images._tag_exists(image_id_or_name, image_obj.tags)
-            ):
-                return image_obj
+        docker = _get_executable("docker")
+        try:
+            result = subprocess.run(
+                [*docker, "inspect", image_id_or_name, "--type", "image"],
+                check=True,
+                capture_output=True,
+                text=True,
+            )
+            json_dict = json.loads(result.stdout)
+        except subprocess.CalledProcessError as ex:
+            raise ImageNotFound(f"Image {image_id_or_name} not found.") from ex
```

Comment:
Couldn't in principle other stuff (and not ImageNotFound) be the source of this error? Like docker not installed or similar (although this specific example I made is validated inside sdk.config.validate_executable). Probably fine for now, however 👍

Reply:
Let's revisit if this becomes a problem in practice. Yes the pattern of obscuring errors and replacing them with out best bet at what went wrong is a bit questionable, but should be OK if power users / devs have a way to opt out and see the original (raw) error message (e.g. via `--loglevel debug`). The latter needs some more work but is also a more general issue not addressed here.